README.md

```diff
@@ -78,37 +79,61 @@ You might not have to sort the output, if you run through reads
 
 Implement the two algorithms in two tools, `naive` and `lin`, that must be present at the root of the repository once they are built. The test setup checks that they give the correct output on selected data, but you should still carefully test them.
 
-Once you have implemented the tools, fill out the report below. 
+Once you have implemented the tools, fill out the report below.
 
 ## Report
 
-### Insights you may have had while implementing and comparing the algorithms. 
+### Insights you may have had while implementing and comparing the algorithms.
 
-*Describe this here.*
+Implementing the naïve algorithm was straightforward, and so should have been the linear one, for which we chose KMP, but during the initial coding phase we confused `i` and `j` in some places, which caused a few headscratching moments.
 
-### Problems encountered if any. 
+### Problems encountered if any.
 
-*Describe this here.*
+Our own testing revealed a flaw in constructing border arrays, meaning that in some cases, when two matches overlapped in the string by a single character, we would not recognize the second match. To add insult to injury, this was already the second, improved version of calculating the border arrays, after a way too simple approach on the first try.
```

Comment:
I'm not sure where the first attempt went wrong, but great to hear that you figured it out and that it is now working.